src/azure-cli/azure/cli/command_modules/monitor/_exception_handler.py

```diff
@@ -4,61 +4,16 @@
-    from azure.mgmt.monitor.v2018_03_01.models import ErrorResponseException as ErrorResponseException_v2018_03_01
-    from azure.mgmt.monitor.v2019_06_01.models import ErrorResponseException as ErrorResponseException_v2019_06_01
-    from msrestazure.azure_exceptions import CloudError
-
-    from knack.util import CLIError
-
-    if isinstance(ex, (ErrorResponseException_v2015_04_01,
-                       ErrorResponseException_v2016_03_01,
-                       ErrorResponseException_v2017_04_01,
-                       ErrorResponseException_v2017_05_01,
-                       ErrorResponseException_v2018_01_01,
-                       ErrorResponseException_v2018_03_01,
-                       ErrorResponseException_v2019_06_01,
-                       CloudError
-                       )):
-        # work around for issue: https://github.com/Azure/azure-sdk-for-python/issues/1556
-        try:
-            error_payload = ex.response.json()
-        except ValueError:
-            raise CLIError(ex)
-        error_payload = {k.lower(): v for k, v in error_payload.items()}
-        if 'error' in error_payload:
-            error_payload = error_payload['error']
-        if 'code' in error_payload and 'message' in error_payload:
-            message = '{}.'.format(error_payload['message']) if error_payload['message'] else 'Operation failed.'
-            code = '[Code: "{}"]'.format(error_payload['code']) if error_payload['code'] else ''
-            raise CLIError('{} {}'.format(message, code))
-        raise CLIError(ex)
-    import sys
-    from six import reraise
-    reraise(*sys.exc_info())
-
-
-def missing_resource_handler(exception):
+def exception_handler(ex):
     from msrest.exceptions import HttpOperationError
-    from knack.util import CLIError
-
-    if isinstance(exception, HttpOperationError) and exception.response.status_code == 404:
-        raise CLIError('Can\'t find the resource.')
-    raise CLIError(exception.message)
-
-
-def data_export_handler(ex):
     from azure.mgmt.loganalytics.models import DataExportErrorResponseException
-    from knack.util import CLIError
-    if isinstance(ex, (DataExportErrorResponseException)):
+    if isinstance(ex, (DataExportErrorResponseException,)):
         ex.message = ex.response.text
-        raise CLIError(ex)
-    import sys
-    from six import reraise
-    reraise(*sys.exc_info())
+    elif isinstance(ex, HttpOperationError):
+        # work around for issue: https://github.com/Azure/azure-sdk-for-python/issues/1556
+        additional_properties = getattr(ex.error, 'additional_properties', {})
+        if 'Code' in additional_properties and 'Message' in additional_properties:
+            ex.error.code = additional_properties['Code']
+            ex.error.message = additional_properties['Message']
+            ex.message = additional_properties['Message']
+    raise ex
```

Comment:
>ex [](start = 10, length = 2)

Is it better that we throw our own exception which wraps the actual exception from SDK?

Reply:
Hunkun has already implemented an universal exception wrapper so it's better to use it.